OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -296,6 +296,14 @@ public WDist GetMaximumRangeVersusTarget(Target target)
 		// Enumerates all armaments, that this actor possesses, that can be used against Target t
 		public IEnumerable<Armament> ChooseArmamentsForTarget(Target t, bool forceAttack)
 		{
+			// If the target is frozen, make sure it is still valid
+			if (t.Type == TargetType.FrozenActor)
+			{
+				var frozenLayer = self.Owner.PlayerActor.TraitOrDefault<FrozenActorLayer>();
+				if (frozenLayer == null || frozenLayer.FromID(t.FrozenActor.ID) == null)
```

Comment:
Your request is problematic. Let's say that target belongs to player B, and it is attacked by players A and C (they are not allies, it's ffa). Let's say that A managed to expose the shroud and see that the target already died, but C didn't, then the target should be `TargetType.Invalid` for A but `TargetType.FrozenActor` for C. I do not think that Target objects should also contain information about who is attacking them, am I correct? 

Do you have another solution or can I leave this code as it is?